src/frontend/src/handler/create_sink.rs

```diff
@@ -656,23 +687,30 @@ pub(crate) async fn reparse_table_for_sink(
     Ok((graph, table, source))
 }
 
-pub(crate) fn insert_merger_to_union(node: &mut StreamNode) {
+pub(crate) fn insert_merger_to_union_with_project(
+    node: &mut StreamNode,
+    project_node: &PbNodeBody,
+    uniq_name: &str,
+) {
     if let Some(NodeBody::Union(_union_node)) = &mut node.node_body {
         node.input.push(StreamNode {
-            identity: "Merge (sink into table)".to_string(),
-            fields: node.fields.clone(),
-            node_body: Some(NodeBody::Merge(MergeNode {
-                upstream_dispatcher_type: DispatcherType::Hash as _,
+            input: vec![StreamNode {
+                node_body: Some(NodeBody::Merge(MergeNode {
+                    ..Default::default()
+                })),
                 ..Default::default()
```

Comment:
It might be clear to explicitly populate these defaults value and add an todo comment here? https://github.com/risingwavelabs/risingwave/issues/17658